checker.cpp

```diff
@@ -2,21 +2,54 @@
 #include <iostream>
 using namespace std;
 
-bool batteryIsOk(float temperature, float soc, float chargeRate) {
-  if(temperature < 0 || temperature > 45) {
-    cout << "Temperature out of range!\n";
-    return false;
-  } else if(soc < 20 || soc > 80) {
-    cout << "State of Charge out of range!\n";
-    return false;
-  } else if(chargeRate > 0.8) {
-    cout << "Charge Rate out of range!\n";
-    return false;
+#define Temperature_Min  0
+#define Temperature_Max  45
+#define SOC_Min  20
+#define SOC_Max  80
+#define ChargeRate_Min  0.0
+#define ChargeRate_Max  0.8
+/* BMS : Battery Management System */
+enum BMS_Parameters {Temperature,Sate_of_Charge,Charge_Rate};
+const char* BMS_Parameters_Name[] = { "Temperature" ,"SOC" , "ChargeRate" };
+
+// enum BMS_Parameters_Min_Value {Temperature_Min = 0,  SOC_Min = 20, ChargeRate_Min = 0.0};
+// enum BMS_Parameters_Max_Value {Temperature_Max = 45, SOC_Max = 80, ChargeRate_Max = 0.8};
+
+class BMS{
+  private:
+  // Variable to verify BMS OK or Not_OK //
+    bool BMS_OK;
```

Comment:
Nice way to latch true unless out of range! Avoids complexity of and-ing a lot of conditions